packages/ember-glimmer/lib/component-managers/custom.ts

```diff
@@ -52,125 +139,129 @@ export interface ComponentArguments<T = {}> {
   * `update()` - invoked when the arguments passed to a component change
   * `getContext()` - returns the object that should be
 */
-export default class CustomComponentManager<T> extends AbstractComponentManager<
-  CustomComponentState<T> | null,
-  DefinitionState
-> {
-  constructor(private delegate: CustomComponentManagerDelegate<T>) {
+export default class CustomComponentManager<ComponentInstance = Opaque>
+  extends AbstractComponentManager<CustomComponentState<ComponentInstance>, DefinitionState>
+  implements
+    WithStaticLayout<
+      CustomComponentState<ComponentInstance>,
+      DefinitionState<ComponentInstance>,
+      OwnedTemplateMeta,
+      RuntimeResolver
+    > {
+  constructor(private delegate: ManagerDelegate<ComponentInstance>) {
```

Comment:
I am not actually sure if we need this to be a stateful object. The "definition state" and "custom component state" already has the "delegate" in it so we don't have to store it again on `this`. Then this would just be stateless and can be stored in a constant. It's one fewer allocation per component, not sure if it matters much.